sdk/identity/Azure.Identity/src/ClientCertificateCredentialOptions.cs

```diff
@@ -12,16 +12,20 @@ public class ClientCertificateCredentialOptions : TokenCredentialOptions, IToken
         /// <summary>
         /// If set to true the credential will store tokens in a cache persisted to the machine, protected to the current user, which can be shared by other credentials and processes.
         /// </summary>
-        public bool EnablePersistentCache { get; set; }
+        internal bool EnablePersistentCache { get; set; }
 
         /// <summary>
         /// If set to true the credential will fall back to storing tokens in an unencrypted file if no OS level user encryption is available.
         /// </summary>
-        public bool AllowUnencryptedCache { get; set; }
+        internal bool AllowUnencryptedCache { get; set; }
 
         /// <summary>
         /// Will include x5c header to enable subject name / issuer based authentication for the <see cref="ClientCertificateCredential"/>.
         /// </summary>
         public bool IncludeX5CCliamHeader { get; set; }
+
+        bool ITokenCacheOptions.EnablePersistentCache => EnablePersistentCache;
```

Comment:
Would this let somebody partially use the feature?  Should we throw NotImplementedException instead?  I don't feel strongly.

Reply:
Oh, no, it's an internal interface.

Reply:
This is still needed for the implementation. The interface is internal so it still cannot be called explicitly by users.